tests/test_aeo_dataset_registration.py

```diff
@@ -86,25 +91,100 @@ def replace_dataset_path(dataset_config_file, dataset_path):
+
+        logger.info("2. with unexpected col: ")
+        _modify_data_file(data_dir, export_index=True)
+        with pytest.raises(
+            DSGInvalidDimension, match=r"column.*is not expected or of a known dimension type"
+        ):
+            _test_dataset_registration(make_test_project_dir, data_dir, dataset)
+
+        logger.info("3. with a duplicated dimension: ")
+        _modify_data_file(data_dir, duplicate_col="subsector")
+        with pytest.raises((ValueError, DSGInvalidDimension)):
+            # (ValueError,  match=r"*is not a valid DimensionType"),
+            # (DSGInvalidDimension,  match=r"column.*is not expected or of a known dimension type")
+            _test_dataset_registration(make_test_project_dir, data_dir, dataset)
+
+        logger.info("4. with a duplicated pivot col: ")
+        _modify_data_file(data_dir, duplicate_col="elec_heating")
+        with pytest.raises(
+            DSGInvalidDimension, match=r"column.*is not expected or of a known dimension type"
+        ):
+            _test_dataset_registration(make_test_project_dir, data_dir, dataset)
+
+        logger.info("5. missing (non-time) dimension combo: ")
+        _modify_data_file(data_dir, remove_geography_subsector=("pacific", "warehouse"))
+        with pytest.raises(
+            DSGInvalidDataset,
+            match=r"load_data records do not match dimension records for dimension combinations",
+        ):
+            _test_dataset_registration(make_test_project_dir, data_dir, dataset)
+
+        logger.info("6. missing time/dimension combo: ")
+        _modify_data_file(data_dir, drop_first_row=True)
+        if "Growth_Factors" in dataset:
+            msg = r"load_data records do not match dimension records for dimension combinations"
+        else:
+            msg = r"One or more arrays do not have.*timestamps"
+        with pytest.raises(DSGInvalidDataset, match=msg):
+            _test_dataset_registration(make_test_project_dir, data_dir, dataset)
+
+
+def _test_dataset_registration(make_test_project_dir, data_dir, dataset):
+    with TemporaryDirectory() as tmpdir:
+        base_dir = Path(tmpdir)
+        logger.info(f"temp_registry created: {base_dir}")
+        manager = make_registry_for_aeo(
+            base_dir,
+            make_test_project_dir,
+            dataset,
+            dataset_path=data_dir,
+            include_datasets=False,
+        )
+
+
+def _modify_data_file(
+    data_dir,
+    duplicate_col=None,
+    drop_first_row=False,
+    remove_geography_subsector=None,
+    export_index=False,
+):
+    df_data = pd.read_csv(data_dir / "load_data_original.csv")
+
+    if duplicate_col is not None:
+        df_data[f"{duplicate_col}_dup"] = df_data[duplicate_col]
+        df_data = df_data.rename(columns={f"{duplicate_col}_dup": duplicate_col})
+    if remove_geography_subsector is not None:
+        if type(remove_geography_subsector) != tuple:
```

Comment:
I wouldn't bother with a type check. It will fail in a similar way if someone passes something else in.